### DISCUSSION TITLE
Creazione automazione su Home Assistant con Sonoff TH16 e Sonoff MINI

### DISCUSSION THREAD

**Discussion by lnrd00** [indomus/forum#2869] [HUB: Home Assistant]
Salve a tutti, avrei bisogno di una mano per comprendere come realizzare un'automazione all'interno di Home Assistant.
Ho una lampadina infrarossi per riscaldare all'interno di una piccola serra, connessa ad un Sonoff MINI. Sempre all'interno della stessa serra, ho un Sonoff TH16 per il monitoraggio di temperatura e umidità (ma non lo uso per comandare direttamente la lampadina). 
Vorrei creare un'automazione dove, se la temperatura rivelata dal TH16 dovesse andare sotto un certo valore, allora la lampadina si accende, ovvero avviene uno switch del Sonoff MINI.
Ho difficoltà a scrivere il codice, soprattutto non saprei come tradurre in codice le seguenti situazioni:

- Se la lampadina è già accesa quando si scende sotto la soglia di temperatura, magari perché è stata accesa manualmente in precedenza, allora non deve avvenire lo switch del Sonoff Mini, altrimenti questo comporterebbe lo spegnimento della lampadina.
- Vorrei anche che la lampadina si spegnesse nel caso in cui questa soglia venisse superata, cioè quando la temperatura si alza. Pure stavolta ci sarebbe da tenere in considerazione l'eventualità che la lampadina sia stata spenta "manualmente" (ciò può avvenire sia da HA ma anche da un interruttore che è dentro alla serra e che è connesso al Sonoff Mini).

Se a qualcuno interessa provare a cimentarsi nella spiegazione, allora l'ID dell'entità del sensore di temperatura è `sensor.temperatura_serra` mentre l'ID dell'entità del Sonoff MINI che comanda la lampadina è `switch.serra`.
Il valore della soglia può essere uno qualsiasi, ad esempio 3 gradi.

Grazie tanto in anticipo!!

**Comment by Scialla** [comment]
Niente di più semplice ma, devi arrivarci da solo, altrimenti non imparerai mai....

Due dritte:

se la lampada è già accesa quando l'automazione la accende, non succede nulla, rimane accesa, basta usare il servizio `switch.turn_on`
se la lampada è già spenta quando l'automazione la spegne, non succede nulla, rimane spenta, basta usare il servizio `switch.turn_off`
il problema si pone se utilizzi il servizio `switch.toggle` che, se è accesa la spegne e se è spenta la accende..

Quindi, automazione semplicissima, se la temperatura è <= a x, utilizzi il servizio `switch.turn_on` e la accendi
se la temperatura è >= a y, utilizzi il servizio `switch.turn_off` e la spegni

Inizia a farti due automazioni separate per imparare e poi, una volta capito, ne basterà una sola con due trigger e un template nella action

Guardati le guide sul sito per imparare
